### ISSUE TITLE
Ruff format conflict warnings are shown twice

### ISSUE THREAD

**Issue by yakMM** [astral-sh/ruff#8271]
Happening since ruff format stable was introduced afaik.

Tested on ruff `0.1.3`:

**Reproduction steps:**

Create a python project with ruff, minimal config:

```python
[tool.ruff]
select = ["ISC001"]
```

Run `ruff format src`

The warning is showing twice in the console:

```
warning: The following rules may cause conflicts when used with the formatter: `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
warning: The following rules may cause conflicts when used with the formatter: `ISC001`. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
```

**Comment by thernstig** [comment]
I have the same problem. But for another rule. But what I found peculiar is that it does not s how which two rules conflict.

```sh
> ruff format --check .
warning: The following rules may cause conflicts when used with the formatter: 'D206'. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
warning: The following rules may cause conflicts when used with the formatter: 'D206'. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
```

This is my Ruff config:

```toml
[tool.ruff]
cache-dir = "/tmp/.cache/ruff"
line-length = 120

# All uncommented lines should be fixed and enabled
select = [
  # These are a copy of the Ruff defaults
  # "F" is Pyflakes, "E" is pycodestyle
  "E4", "E7", "E9", "F",
  "A", # flake8-builtins
  # "B", # flake8-bugbear
  # "C4", # flake8-comprehensions
  "C90", # mccabe
  # "COM", # flake8-commas
  "D", # pydocstyle
  # "FIX", # flake8-fixme
  # "RUF100", # Ruff rule: Enforce that noqa directives are "valid"
  # "S", # flake8-bandit
  "TID", # flake8-tidy-imports
  # "Q", # flake8-quotes
]

# All these ignores should be fixed (except for Q000 which should stay)
ignore = [
  "COM812", # flake8-commas "Trailing comma missing"
  "D10",  # pydocstyle "Missing docstring in ..."
  "D203", # pydocstyle "one-blank-line-before-class" conflicting with D211
  "D205", # pydocstyle "1 blank line required between summary line and description"
  "D208", # pydocstyle "Docstring is over-indented"
  "D209", # pydocstyle "Multi-line docstring closing quotes should be on a separate line"
  "D211", # pydocstyle "no-blank-line-before-class" conflicting with D203
  "D212", # pydocstyle "multi-line-summary-second-line" conflicting with D213
  "D213", # pydocstyle "no-blank-line-before-class" conflicting with D212
  "D400", # pydocstyle "First line should end with a period"
  "D401", # pydocstyle "First line of docstring should be in imperative mood"
  "D404", # pydocstyle "First word of the docstring should not be "This""
  "D403", # pydocstyle "First word of the first line should be capitalized"
  "D407", # pydocstyle "Missing dashed underline after section ("Return")"
  "D415", # pydocstyle "First line should end with a period, question mark, or exclamation point"
  "D419", # pydocstyle "Docstring is empty"
  "E713", # pycodestyle "Test for membership should be `not in`"
  "Q000", # flake8-quotes "Single quotes found but double quotes preferred"
]

[tool.ruff.format]
quote-style = "single"

[tool.ruff.flake8-tidy-imports]
ban-relative-imports = "all"
```

**Comment by MichaReiser** [comment]
Thanks. I think that happens if you have multiple configurations in your project. We should only show it once.

**Comment by thernstig** [comment]
> Thanks. I think that happens if you have multiple configurations in your project. We should only show it once.

Do you mean multiple `.toml` files? I only have one, see above.

**Comment by MichaReiser** [comment]
@thernstig I would expect this to happen if you have multiple `ruff.toml` or `pyproject.toml`. But maybe there's another case where we verify the same configuration twice.

**Comment by thernstig** [comment]
Got it, I can confirm I only have one.

**Comment by yakMM** [comment]
> @thernstig I would expect this to happen if you have multiple `ruff.toml` or `pyproject.toml`. But maybe there's another case where we verify the same configuration twice.

I have one too. Here is a minimal repo reproducing the bug: https://github.com/yakMM/test-double-warning

**Comment by tylerlaprade** [comment]
This still happens for the `--preview` warning:
<img width="649" alt="image" src="https://github.com/astral-sh/ruff/assets/5475199/647a8e91-4bc9-4cf5-a6d2-cb2071309d49">

**Comment by charliermarsh** [comment]
👍 Yeah that's a separate code path and issue.

**Comment by AmineDjeghri** [comment]
I have the same warning with `D206` as @thernstig 

```
warning: The following rules may cause conflicts when used with the formatter: 'D206'. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
warning: The following rules may cause conflicts when used with the formatter: 'D206'. To avoid unexpected behavior, we recommend disabling these rules, either by removing them from the `select` or `extend-select` configuration, or adding then to the `ignore` configuration.
```

my pre-commit-config.yaml : 

```
# See https://pre-commit.com for more information
# See https://pre-commit.com/hooks.html for more hooks
# ruff is for python files and is used with ruff.toml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.3.0
    hooks:
      - id: check-added-large-files
        args: ['--maxkb=3000']
      - id: check-toml
      - id: check-yaml
      - id: check-json
      - id: detect-private-key
      - id: end-of-file-fixer
      #      - id: requirements-txt-fixer
      #        exclude: ^requirements/.*$
      - id: trailing-whitespace
      - id: name-tests-test
        args: [ --pytest-test-first ]

  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.1.2
    hooks:
      - id: ruff
        args: ['--fix']
      - id: ruff-format

```

my ruff.toml : 

```
extend-include = ["*.ipynb"]
line-length = 100

# read more here https://docs.astral.sh/ruff/settings/#lintpydocstyle
[lint]
# Enable all `pydocstyle` rules, limiting to those that adhere to the Google convention via `convention = "google"`
select = [
    "D",
]
# On top of the Google convention, disable for ex `D417`, which requires documentation for every function parameter.
ignore = ["D100", "D103", "D417"]

[lint.pydocstyle]
# Use Google-style docstrings.
convention = "google"
```

**Comment by zanieb** [comment]
@AmineDjeghri are you just seeing this once for the linter and once for the formatter? Those pre-commit runs are fully separate from our perspective.

**Comment by AmineDjeghri** [comment]
it's fixed in the latest version `v0.4.2` ! 
In my pre-commit-config.yaml, i forgot to update ruff
